StreamChatClient_v3/WebSocketClient/Events/Event.swift

```diff
@@ -5,8 +5,44 @@
 
 import Foundation
 
-/// An `Event` object represent an event in the chat system.
+/// An `Event` object representing an event in the chat system.
 public protocol Event {
     /// The underlying raw type of the incoming string.
     static var eventRawType: String { get }
 }
+
+/// The DTO object mirroring the JSON representation of an event.
+struct EventPayload<ExtraData: ExtraDataTypes>: Decodable {
+    let eventType: String
+    
+    let connectionId: String?
+    
+    let channelPayload: ChannelDetailPayload<ExtraData>?
+    
+    let currentUserPayload: UserPayload<ExtraData.User>? // TODO: Create CurrentUserPayload?
+    
+    let cid: String?
+    
+    private enum CodingKeys: String, CodingKey {
+        case connectionId = "connection_id"
+        case channelPayload = "channel"
+        case eventType = "type"
+        case currentUserPayload = "me"
+        case cid
+    }
+}
+
+/// An internal protocol marking the Events carrying the payload. This payload can be then used for additional work,
+/// i.e. for storing the data to the database.
+protocol EventWithPayload {
+    /// A type-erased `EventPayload`
+    var eventPayload: AnyEventPayload { get }
+}
+
+/// A type-erased wrapper for EventPayload.
+struct AnyEventPayload {
+    let payload: Any
+    init<ExtraData>(_ eventPayload: EventPayload<ExtraData>) {
+        payload = eventPayload as Any
+    }
+}
```

Comment:
Additionally for above questions. What do you think if we will use protocols here. We can do the same eraser but with some `EventPayloadProtocol`. For me it looks like a lot of work but the idea is to erase only extra data types.